cirq/linalg/decompositions_test.py

```diff
@@ -217,7 +207,72 @@ def test_kak_canonicalize_vector(x, y, z):
     cirq.testing.random_unitary(4)
     for _ in range(10)
 ])
-def test_kak_decomposition(m):
-    g, (a1, a0), (x, y, z), (b1, b0) = cirq.kak_decomposition(m)
-    m2 = recompose_kak(g, (a1, a0), (x, y, z), (b1, b0))
-    assert np.allclose(m, m2)
+def test_kak_decomposition(target):
+    kak = cirq.kak_decomposition(target)
+    np.testing.assert_allclose(cirq.unitary(kak), target, atol=1e-8)
+
+
+def test_kak_decomposition_eq():
+    eq = cirq.testing.EqualsTester()
+
+    eq.make_equality_group(lambda: cirq.KakDecomposition(
+        global_phase=1,
+        single_qubit_operations_before=(cirq.unitary(cirq.X),
+                                        cirq.unitary(cirq.Y)),
+        interaction_coefficients=(0.3, 0.2, 0.1),
+        single_qubit_operations_after=(np.eye(2), cirq.unitary(cirq.Z)),
+    ))
+
+    eq.add_equality_group(cirq.KakDecomposition(
+        global_phase=-1,
+        single_qubit_operations_before=(cirq.unitary(cirq.X),
+                                        cirq.unitary(cirq.Y)),
+        interaction_coefficients=(0.3, 0.2, 0.1),
+        single_qubit_operations_after=(np.eye(2), cirq.unitary(cirq.Z)),
+    ))
+
+    eq.make_equality_group(lambda: cirq.KakDecomposition(
+        global_phase=1,
+        single_qubit_operations_before=(cirq.unitary(cirq.X),
+                                        cirq.unitary(cirq.H)),
+        interaction_coefficients=(0.3, 0.2, 0.1),
+        single_qubit_operations_after=(np.eye(2), cirq.unitary(cirq.Z)),
+    ))
+
+    eq.make_equality_group(lambda: cirq.KakDecomposition(
+        global_phase=1,
+        single_qubit_operations_before=(cirq.unitary(cirq.X),
+                                        cirq.unitary(cirq.Y)),
+        interaction_coefficients=(0.5, 0.2, 0.1),
+        single_qubit_operations_after=(np.eye(2), cirq.unitary(cirq.Z)),
+    ))
```

Comment:
This just tests that all of these are not equal, right?  Should we have a group with more than one element?

Reply:
Note that one of them is a `make_` call instead of `add_`. `make_` invokes the lambda twice and checks the results for equality. It exists exactly for this case, to avoid repetition.